tournamentcontrol/competition/tests/test_logo_fields.py

```diff
@@ -0,0 +1,255 @@
+                )
+            }
+        )
+
+    def test_club_logo_accepts_jpg(self):
+        """Test that Club accepts JPG files for logos via admin."""
+        club = ClubFactory()
+        namespace = club._get_admin_namespace()
+        args = club._get_url_args()
+        
+        self.assertGoodEditView(
+            "%s:edit" % namespace,
+            *args,
+            data={
+                "title": club.title,
+                "slug": club.slug,
+                "logo_monochrome": SimpleUploadedFile(
+                    "club_logo.jpg", self.jpg_data, content_type="image/jpeg"
+                )
+            }
+        )
+
+    def test_season_logo_accepts_svg(self):
+        """Test that Season accepts SVG files for logos via admin."""
+        season = SeasonFactory()
+        namespace = season._get_admin_namespace()
+        args = season._get_url_args()
+        
+        self.assertGoodEditView(
+            "%s:edit" % namespace,
+            *args,
+            data={
+                "title": season.title,
+                "slug": season.slug,
+                "logo_colour": SimpleUploadedFile(
+                    "season_logo.svg", self.svg_data, content_type="image/svg+xml"
+                )
+            }
+        )
+
+    def test_division_logo_accepts_png(self):
+        """Test that Division accepts PNG files for logos via admin."""
+        division = DivisionFactory()
+        namespace = division._get_admin_namespace()
+        args = division._get_url_args()
+        
+        self.assertGoodEditView(
+            "%s:edit" % namespace,
+            *args,
+            data={
+                "title": division.title,
+                "slug": division.slug,
+                "logo_colour": SimpleUploadedFile(
+                    "division_logo.png", self.png_data, content_type="image/png"
+                )
+            }
+        )
+
+    def test_team_logo_accepts_jpg(self):
+        """Test that Team accepts JPG files for logos via admin."""
+        team = TeamFactory()
+        namespace = team._get_admin_namespace()
+        args = team._get_url_args()
+        
+        self.assertGoodEditView(
+            "%s:edit" % namespace,
+            *args,
+            data={
+                "title": team.title,
+                "slug": team.slug,
+                "logo_monochrome": SimpleUploadedFile(
+                    "team_logo.jpg", self.jpg_data, content_type="image/jpeg"
+                )
+            }
+        )
+
+    def test_division_logo_rejects_pdf(self):
+        """Test that Division rejects PDF files."""
+        division = DivisionFactory()
+        pdf_file = SimpleUploadedFile(
+            "division_logo.pdf", self.pdf_data, content_type="application/pdf"
+        )
+        
+        # Assign invalid file and expect validation error
+        division.logo_colour = pdf_file
+        
+        with self.assertRaises(ValidationError) as cm:
+            division.full_clean()
```

Comment:
This should manifest as a failed upload on a POST to edit the logo. Again, test via the views.